stores/map/slice.tsx

```diff
@@ -72,15 +88,20 @@ export const mapSlice = createSlice({
       state.reloadBuildings = Math.random(); // Force le trigger de useEffect
     },
     updateAddresses(state, action) {
-      if (state.selectedBuilding) {
-        state.selectedBuilding.addresses = action.payload;
+      if (state.selectedItem && state.selectedItemType === 'building') {
+        state.selectedItem.addresses = action.payload;
       }
     },
+    unselectItem(state) {
+      state.selectedItemType = undefined;
+      state.selectedItem = undefined;
+    },
   },
 
   extraReducers(builder) {
     builder.addCase(selectBuilding.fulfilled, (state, action) => {
-      state.selectedBuilding = action.payload;
+      state.selectedItemType = 'building';
```

Comment:
Une façon de faire pour détecter si on a un `SelectedBuilding` ou `SelectedADS` est de tester  i on a un champs champ obligatoire présent (ex: `'is_active' in action.payload`) ou d'utiliser un discriminant dans l'interface:

```
export interface SelectedBuilding {
  _type: 'bulding'
  // Les autres champs
}

export interface SelectedADS {
  _type: 'ads'
  // Les autres champs
}
```

Puis de tester avec: `action.payload._type === 'building'` (`_type` est à ajouter dans l'objet lors de la récupération de la donnée lors de l'appel REST).

Reply:
Ah super, je n'étais pas très content de mon système avec deux variables différentes.